### PR TITLE
Update some logging in the project system

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Just logging more things next time we need it.

### CODE REVIEW

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LoadedProject.cs [hunk 1]
```diff
@@ -108,69 +112,78 @@ public async ValueTask<(ImmutableArray<CommandLineReference>, OutputKind)> Updat
...
-            reference => _projectSystemProject.RemoveAnalyzerReference(reference.FilePath));
+            reference => _projectSystemProject.RemoveAnalyzerReference(reference.FilePath),
+            "Project {0} now has {1} analyzer reference(s).");
 
         _mostRecentAnalyzerReferences = analyzerReferences;
 
         UpdateProjectSystemProjectCollection(
-            newProjectInfo.AdditionalDocuments.Distinct(DocumentFileInfoComparer.Instance), // TODO: figure out why we have duplicates
-            _mostRecentFileInfo?.AdditionalDocuments.Distinct(DocumentFileInfoComparer.Instance),
+            newProjectInfo.AdditionalDocuments,
```
Comment (dibarbet) [inline src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LoadedProject.cs:144]: is the todo comment resolved by something else?

**Review by dibarbet — APPROVED** [review]

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LoadedProject.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LoadedProject.cs:144]: It was resolved by me realizing that sometimes folks just have their MSBuild targets adding an additional file twice, so it's not an interesting thing to chase. :smile:

**Pull request merged**